diffxpy/testing/base.py

```diff
@@ -1310,6 +1318,339 @@ def summary_pair(self, group1, group2,
+        """
+        if groups0 is None:
+            groups0 = self.groups
+        if groups1 is None:
+            groups1 = self.groups
+        self._check_groups(groups0, groups1)
+        groups0 = self._groups_idx(groups0)
+        groups1 = self._groups_idx(groups1)
+
+        num_features = self._theta_mle.shape[1]
+
+        logfc = np.zeros(shape=(len(groups0), len(groups1), num_features))
+        for i,g0 in enumerate(groups0):
+            for j,g1 in enumerate(groups1):
+                logfc[i,j,:] = self._theta_mle[g0,:].values - self._theta_mle[g1,:].values
+
+        if base == np.e:
+            return logfc
+        else:
+            return logfc / np.log(base)
+
+    def summary_pair(self, group0, group1,
+                     qval_thres=None, fc_upper_thres=None,
+                     fc_lower_thres=None, mean_thres=None,
+                     **kwargs) -> pd.DataFrame:
+        """
+        Summarize differential expression results of single pairwose comparison
+        into an output table.
+
+        :param group0: Firt group in pair-wise comparison.
+        :param group0: Second group in pair-wise comparison.
+        :return: pandas.DataFrame with the following columns:
+
+            - gene: the gene id's
+            - pval: the per-gene p-value of the selected test
+            - qval: the per-gene q-value of the selected test
+            - log2fc: the per-gene log2 fold change of the selected test
+            - mean: the mean expression of the gene across all groups
+        """
+        assert self.gene_ids is not None
+
+        if len(group0) != 1:
+            raise ValueError("group0 should only contain one entry in summary_pair()")
+        if len(group1) != 1:
+            raise ValueError("group1 should only contain one entry in summary_pair()")
+
+        pval = self.pval_pairs(groups0=group0, groups1=group1)
+        qval = self._correction(pvals=pval, **kwargs)
+        res = pd.DataFrame({
+            "gene": self.gene_ids,
+            "pval": pval.flatten(),
+            "qval": qval.flatten(),
+            "log2fc": self.log_fold_change_pairs(groups0=group0, groups1=group1, base=2).flatten(),
+            "mean": np.asarray(self.mean)
+        })
+
+        res = self._threshold_summary(
+            res=res,
+            qval_thres=qval_thres,
+            fc_upper_thres=fc_upper_thres,
+            fc_lower_thres=fc_lower_thres,
+            mean_thres=mean_thres
+        )
+
+        return res
+
+    def summary_pairs(self, groups0, groups1=None,
+                     qval_thres=None, fc_upper_thres=None,
+                     fc_lower_thres=None, mean_thres=None,
+                     **kwargs) -> pd.DataFrame:
+        """
+        Summarize differential expression results of a set of
+        pairwise comparisons into an output table.
+
+        :param groups0: First set of groups in pair-wise comparison.
+        :param groups0: Second set of groups in pair-wise comparison.
```

Comment:
2x `groups0` => typo?

Reply:
Fixed this and other occurences in https://github.com/theislab/diffxpy/commit/17c77184523bf8238a93eff7e5117d27a2b85fd6.